app/validation.py

```diff
@@ -176,3 +178,17 @@ def is_valid_service_id_or_400(service_id):
         return True
     else:
         abort(400, "Invalid service ID supplied: %s" % service_id)
+
+
+def is_valid_date(date, default_format='%Y-%m-%d'):
+    try:
+        datetime.strptime(date, default_format)
+        return True
+    except ValueError:
+        return False
+
+
+def is_valid_acknowledged_state(acknowledged):
+    if acknowledged in ['all', 'acknowledged', 'not-acknowledged']:
+        return True
+    return False
```

Comment:
Really nitpicky, but this could just be `return acknowledged in ['all', 'acknowledged', 'not-acknowledged']`